From b49ab7a07498ecb595045446448912c171dfa991 Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Tue, 9 Apr 2019 16:55:28 +0200
Subject: [PATCH] Do not create/destroy Windows device on Primary/Secondary.

We do this - for now - in setting the mount point of the
device (via windrbd set-mount-point-for-minor), this currently
fails with no such device (but this is maybe a mountmanager
bug).
---
 drbd/drbd_nl.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/drbd/drbd_nl.c b/drbd/drbd_nl.c
index 06c5a32..71f3069 100644
--- a/drbd/drbd_nl.c
+++ b/drbd/drbd_nl.c
@@ -1037,14 +1037,6 @@ drbd_set_role(struct drbd_resource *resource, enum drbd_role role, bool force, s
 	enum chg_state_flags flags = CS_ALREADY_SERIALIZED | CS_DONT_RETRY | CS_WAIT_COMPLETE;
 	enum drbd_role old_role = resource->role[NOW];
 
-	if (old_role == R_PRIMARY && role == R_SECONDARY) {
-		idr_for_each_entry(&resource->devices, device, vnr) {
-			if (windrbd_umount(device->this_bdev) != 0)
-				windrbd_device_error(device, &err_str, "Warning: couldn't umount volume %d\n", vnr);
-			windrbd_remove_windows_device(device->this_bdev);
-		}
-	}
-
 retry:
 	down(&resource->state_sem);
 
@@ -1273,15 +1265,6 @@ retry:
 			drbd_kobject_uevent(device);
 	}
 
-	if (old_role == R_SECONDARY && role == R_PRIMARY) {
-		idr_for_each_entry(&resource->devices, device, vnr) {
-			if (windrbd_create_windows_device(device->this_bdev) != 0)
-				windrbd_device_error(device, &err_str, "Warning: Couldn't create windows device for volume %d\n", vnr);
-			if (windrbd_mount(device->this_bdev) != 0)
-				windrbd_device_error(device, &err_str, "Warning: Couldn't mount volume %d, perhaps the drive letter (%S) is in use?\n", vnr, device->this_bdev->mount_point.Buffer);
-		}
-	}
-
 out:
 	up(&resource->state_sem);
 	if (err_str) {
-- 
2.17.0

